runtime/stdlib.js

```diff
@@ -352,85 +352,129 @@ function caml_compare_val (a, b, total) {
   var stack = [];
   for(;;) {
     if (!(total && a === b)) {
```

Comment:
What is "total"?

Reply:
`total` say whether we need a total order or not.
We only need a total order for polymorphic compare, not for boolean operator  `=`, `<>`, `<`, `>`, ...

Reply:
For instance, `compare nan nan = 0` but `nan <> nan`,`(nan = nan) = false`, `(nan < nan) = false`, `(nan > nan) = false`